src/BuildPrediction/Predictors/ReferenceItemsPredictor.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Build.Prediction.Predictors
+{
+    using System;
+    using System.IO;
+    using Microsoft.Build.Evaluation;
+    using Microsoft.Build.Execution;
+
+    /// <summary>
+    /// Finds EmbeddedResource items as inputs.
+    /// </summary>
+    public class ReferenceItemsPredictor : IProjectPredictor
+    {
+        internal const string ReferenceItemName = "Reference";
+
+        internal const string HintPathMetadata = "HintPath";
+
+        private readonly char[] _invalidPathCharacters = Path.GetInvalidPathChars();
```

Comment:
>  [](start = 15, length = 1)

static

Reply:
Will add comment as to why this isn't static